### ISSUE TITLE
$mergeObjects support

### ISSUE THREAD

**Issue by miguelalvarezi** [mongomock/mongomock#642]
First of all, thanks for this great library!

I'm trying to test an aggregation pipeline which includes a stage like the following:

```
{'$replaceRoot': {'newRoot': {'$mergeObjects': [default_doc, '$_id']}}}
```

I'm getting the following error:
 
```
pymongo.errors.OperationFailure: Unrecognized expression '$mergeObjects'
```

I suppose the library doesn't support `$mergeObjects` yet. Am I right?

Thanks!

**Comment by pcorpet** [comment]
You're right it's not implemented. Although I think it would be easy to add if you want to send a PR.

I'm fixing the "bug" that it's raising an OperationFailure instead of a NotImplementedError.
